File: android/src/main/java/com/github/quadflask/react/navermap/RNNaverMapCircleOverlay.java
package com.github.quadflask.react.navermap;

import android.content.Context;

import androidx.annotation.NonNull;
import com.naver.maps.geometry.LatLng;
import com.naver.maps.map.overlay.CircleOverlay;
import com.naver.maps.map.overlay.Overlay;

public class RNNaverMapCircleOverlay extends ClickableRNNaverMapFeature<CircleOverlay> {
    public RNNaverMapCircleOverlay(EventEmittable emitter, Context context) {
        super(emitter, context);
        feature = new CircleOverlay();
    }

    public void setCenter(LatLng center) {
        feature.setCenter(center);
    }

    public void setRadius(double radius) {
        feature.setRadius(radius);
    }

    public void setColor(int color) {
        feature.setColor(color);
    }

    public void setOutlineWidth(int width) {
        feature.setOutlineWidth(width);
    }

    public void setOutlineColor(int color) {
        feature.setOutlineColor(color);
    }

    public void setZIndex(int zIndex) {
        feature.setZIndex(zIndex);
    }

    @Override
    public boolean onClick(@NonNull Overlay overlay) {
        super.onClick(overlay);
        return false;
    }
}

File: android/src/main/java/com/github/quadflask/react/navermap/feature/RNNaverMapCircleOverlay.kt
package com.github.quadflask.react.navermap.feature

import android.content.Context
import com.naver.maps.geometry.LatLng
import com.naver.maps.map.overlay.CircleOverlay

class RNNaverMapCircleOverlay(emitter: EventEmittable, context: Context) : ClickableRNNaverMapFeature<CircleOverlay>(emitter, context) {
  init {
    feature = CircleOverlay()
  }

  fun setCenter(center: LatLng) {
    feature.center = center
  }

  fun setRadius(radius: Double) {
    feature.radius = radius
  }

  fun setColor(color: Int) {
    feature.color = color
  }

  fun setOutlineWidth(width: Int) {
    feature.outlineWidth = width
  }

  fun setOutlineColor(color: Int) {
    feature.outlineColor = color
  }

  fun setZIndex(zIndex: Int) {
    feature.zIndex = zIndex
  }
}
